asdf/tests/test_entry_points.py

```diff
@@ -1,12 +1,6 @@
-import sys
-
+import importlib_metadata as metadata
 import pytest
 
-if sys.version_info < (3, 10):
-    import importlib_metadata as metadata
-else:
-    import importlib.metadata as metadata
-
```

Comment:
I do not like this, we should be using the builtin version of `importlib.metadata` whenever possible. I seems like you need v4.11 of the `importlib-metadata` package due to the bug. Looking at this table: https://github.com/python/importlib_metadata#compatibility, python 3.11 has a new enough version of `importlib-metadata` builtin in order to avoid the bug.

Please change the `version_info` check to `(3, 11)` instead of removing this.